packages/gatsby/src/redux/reducers/telemetry.ts

```diff
@@ -0,0 +1,46 @@
+import { ActionsUnion, IGatsbyState } from "../types"
+
+const initialState = {
+  gatsbyImageSourceUrls: new Set<string>(),
+}
+
+export const telemetryReducer = (
+  state: IGatsbyState["telemetry"] = initialState,
+  action: ActionsUnion
+): IGatsbyState["telemetry"] => {
+  switch (action.type) {
+    case `PROCESS_GATSBY_IMAGE_SOURCE_URL`: {
+      const { sourceUrl } = action.payload
+      const nextState = new Set(state.gatsbyImageSourceUrls)
```

Comment:
Similar thing for `MERGE_WORKER_QUERY_STATE` action

Reply:
Got it! Updated sot that all these actions modify the existing state object instead of returning a new one:

- `PROCESS_GATSBY_IMAGE_SOURCE_URL`
- `CLEAR_GATSBY_IMAGE_SOURCE_URL`
- `MERGE_WORKER_QUERY_STATE`